Android.bp

```diff
@@ -29,7 +29,7 @@ cc_defaults {
         "-DDLT_DAEMON_USE_UNIX_SOCKET_IPC",
         "-DDLT_LIB_USE_UNIX_SOCKET_IPC",
         "-DCONFIGURATION_FILES_DIR=\"/vendor/etc\"",
-        "-DDLT_USER_IPC_PATH=\"/data/local/tmp\"",
+        "-DDLT_USER_IPC_PATH=\"/dev/socket\"",
```

Comment:
Thank you @sebastienraillet 
What do you think about "/data/vendor"?

Reply:
Hello @thanhbnq,
Of my understanding, _/data/vendor_ is more about storing specific data for native applications (e.g internal database, binary files, etc.).

I've use _/dev/socket_ as it is the location where init daemon creates socket declared inside the .rc file (see [here](https://cs.android.com/android/platform/superproject/+/master:system/core/init/util.cpp;drc=3b3d87de79ab993eb1dc03286deb6f6c0e8c7c68;l=89) and [here](https://cs.android.com/android/platform/superproject/+/master:device/generic/vulkan-cereal/fake-android-guest/cutils/sockets.h;drc=9958a2d1d95a4c9ad1578ffe6cc0b78e7368e22b;l=46))

Now for the sake of curiosity 😉, I've tried to move the sockets inside _/data/vendor_ to understand what are the drawbacks in this case. You can find below a summary : 
## sockets inside /dev/socket
* sockets are created and setup (permissions, mode, etc.) by init process
* recovered by dlt-daemon through _**android_get_control_socket()**_ API
* allow to specify one SELinux label for each sockets (this is done by init when sockets are created) 
## sockets inside /data/vendor
* a specific folder inside /data/vendor is created by init (e.g dlt folder)
* sockets are created inside this folder by dlt-daemon as usual
* dlt-daemon sockets will inherit the SELinux label of the folder. Means we can't put different labels on each sockets.

Of my point of view, in term of security, and based on the principale of least privilege, it is better to be able to apply distinct SELinux label for each socket. Like that, you can split easily application having access to dlt app socket and dlt control socket.
By checking also the aosp documentation [here](https://source.android.com/security/selinux/compatibility#file-ownership), they are talking about _/dev/vendor/socket_. I'm not able to find this path neither on Android 10 nor in Android 11. Knowing that this documentation has been updated on 22 septembre 2021, maybe this is something planned for Android 12 🤞 . Means that vendor applications will potentially have a specific location for their sockets in the future.

But maybe you had in mind some corner cases I've not in mind on my side where _/dev/socket_ doesn't sounds a good location 🙂 ?